src/sagas/background-refresh.js

```diff
@@ -0,0 +1,213 @@
+import {
+  all,
+  call,
+  put,
+  race,
+  select,
+  take,
+  takeEvery,
+  takeLatest,
+  throttle,
+} from 'redux-saga/effects'
+
+import * as Actions from '_/actions'
+import * as C from '_/constants'
+
+import AppConfiguration from '_/config'
+import { isNumber } from '_/utils'
+import { delay } from './utils'
+
+import {
+  fetchPoolsByCount,
+  fetchSinglePool,
+  fetchSingleVm,
+  fetchVmsByCount,
+  selectVmDetail,
+} from './index'
+import { getConsoleOptions } from './console'
+import { fetchIsoFiles } from './storageDomains'
+
+/**
+ * Change the background refresh type based on the page type, and force a refresh.
+ *
+ * This should be done at the time of navigation to the page, typically by the page router.
+ */
+function* changePage (action) {
+  yield put(Actions.stopSchedulerFixedDelay())
```

Comment:
Is it necessary to tear down the scheduler? Perhaps it would be enough to force refresh?
With next cycle the scheduler should trigger the refresh for the correct page anyway.
Tearing down the scheduler should not disrupt any logic (it should be prepared for that) but it
makes the scheduler look  one-purpose-only (for refreshing) and  one-page-only.
It seems however that this service will grow in the future into kind of event loop which handles various actions i.e.
in [Add user settings PR](https://github.com/oVirt/ovirt-web-ui/pull/1088) we need to toggle notifications globaly on a time basis.
The proposed implementation re-uses scheduler to accomplish that goal.

Reply:
The scheduler has become single purpose.  It's job is to use polling to refresh useful data for the currently displayed page.

For the user settings side of things, I have no problem setting up other single use timer based tasks.  But we can chat about it on #1088.